crypto/thresholdsign.go

```diff
@@ -109,11 +110,16 @@ func NewThresholdSigner(size int, threshold int, currentIndex int, hashAlgo hash
 // be used as the input keys to this function.
 func (s *thresholdSigner) SetKeys(currentPrivateKey PrivateKey,
 	groupPublicKey PublicKey,
-	sharePublicKeys []PublicKey) {
+	sharePublicKeys []PublicKey) error {
+
+	if len(sharePublicKeys) != s.size {
+		return fmt.Errorf("size of public key shares should be %d", s.size)
+	}
 
 	s.currentPrivateKey = currentPrivateKey
```

Comment:
That's right, same comment also applies to `SetMessage`. On the other hand, we would like to still be able to rotate the keys and message on the same threshold instance. 
Since the final threshold signature is always verified by the api, any error will be caught, but maybe we need a cleaner way to avoid errors.

Reply:
I think I could just call `ClearShares` when setting keys. This erases all existing shares that are signed with older keys.